### PR TITLE
upgrade python versions, add type annotations, drop bad args styles

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
- [x] drop deprecated global navigator
- [x] add type annotations
- [x] deprecate additional magical step arguments (we ought to use objects there)
- [x] deprecate type as navigate targets

### CODE REVIEW

File: src/navmazing/__init__.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    class ShowAllProviders(NavigateStep)
+        def am_i_here(self):
+            return check_if_i_am_already_on_page()
+
+        def step(self):
+            click('All button')
+
+"""
+from __future__ import annotations
+from ._errors import NavigationTriesExceeded
```
Comment (ikerreyes) [inline src/navmazing/__init__.py:31]: What is the advantage of using `_` as prefix if module names? Or is it just for convention?

**Review by ikerreyes — COMMENTED** [review]
Thanks @RonnyPfannschmidt I believe this is a great improvement.

Question about the `py.typed` files. Is it used to force `mypy` to run? If so, could we add comment in the file indicating it?

File: src/navmazing/__init__.py (hunk 1 above)
Comment (RonnyPfannschmidt) [reply src/navmazing/__init__.py:31]: That's the convention for private modules

Other parties ought not to import from those

**Comment by RonnyPfannschmidt** [comment]
The `py.typed` file tells mypy that navmazing has Type annotations others may use, else mypy is considering navmazing as untyped even if annotations are added

File: src/navmazing/_step.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            return
+
+    def post_navigate(self, _tries: int, *args: object, **kwargs: object) -> None:
+        """steps taken after the navigation is complete
+
+        This is a default and is generally overridden.
+        """
+        return
+
+    def go(self, _tries: int = 0, *args: object, **kwargs: object) -> object:
```
[the remark this answers is not in the document]
Comment (ikerreyes) [reply src/navmazing/_step.py:106]: Oh, I see. Thank you!

**Comment by ikerreyes** [comment]
Thanks for the explanations @RonnyPfannschmidt

**Pull request merged**